formatter/src/main/java/org/togetherjava/formatter/tokenizer/Lexer.java

```diff
@@ -0,0 +1,92 @@
+package org.togetherjava.formatter.tokenizer;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Tokenizer that can turn a list of strings (or a string) into a list of tokens
+ *
+ * @author illuminator3
+ */
+public class Lexer {
+    private static final Pattern commentPatcherRegex =
```

Comment:
What does this do? A complicated Regex without any example/explanation is quite hard to understand.